packages/format/format.pony

```diff
@@ -96,8 +96,10 @@ primitive Format
       _FormatInt.u128(x.u128(), false, fmt, prefix, prec, width, align, fill)
     elseif x is I128 then
       _FormatInt.u128(abs.u128(), neg, fmt, prefix, prec, width, align, fill)
-    else
+    elseif (x is U64) or (x is I64) then
```

Comment:
there's a bug that predates your code @krig. These should be using a match expression or an `iftype`. The latter is preferred.

The `is` doesn't work as it compares identity rather than checking the type.

So `x is I128` will only be triggered for the value 0. otherwise the else will always be used.